scipy/stats/tests/test_mstats_basic.py

```diff
@@ -78,14 +78,13 @@ def test_1d_ma_value(self):
     def test_1d_ma0(self):
         #  Test a 1d masked array with zero element
         a = np.ma.array([10, 20, 30, 40, 50, 60, 70, 80, 90, 0])
-        desired = 41.4716627439
```

Comment:
When `np.log` is taken on this masked array, 0 becomes `-np.inf` and gets masked, so the average doesn't consider it, and it behaves just like we had done:

```python3
        a = np.ma.array([10, 20, 30, 40, 50, 60, 70, 80, 90, 100], mask=[0, 0, 0, 0, 0, 0, 0, 0, 0, 1])
```
as in the previous test. 

Is that a "feature"? If so, we need to separate `mstats.gmean` from `stats.gmean`, because I don't think that's how `stats.gmean` should work.